### PR TITLE
JIT: Stop using ref counts in forward sub

### PR DESCRIPTION
**Pull request by jakobbotsch**
Instead of going to great lengths to keep the ref counts correct in physical promotion just stop using them in forward sub. The last use information generalizes what forward sub needs except for when the tracking limit is reached, which shows in the improvements we actually get from consulting the ref counts -- the improvements are very modest and in tests only.

This makes early liveness the only consumer of the "regular" ref counts computed by local morph. Weighted ref counts are still used for undoing regular promotion of implicit byref parameters.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @AndyAyersMS 

[Diffs.](https://dev.azure.com/dnceng-public/public/_build/results?buildId=265649&view=ms.vss-build-web.run-extensions-tab)

Let me know if you have any objections to this...

File: src/coreclr/jit/forwardsub.cpp [hunk 1]
```diff
@@ -495,24 +481,12 @@ bool Compiler::fgForwardSubStatement(Statement* stmt)
         return false;
     }
 
-    // Only fwd sub if we expect no code duplication
+    // Cannot forward sub without liveness information.
     //
-    bool livenessBased = false;
-    if (varDsc->lvRefCnt(RCS_EARLY) != 2)
+    if (!fgDidEarlyLiveness)
```
Comment (AndyAyersMS) [inline src/coreclr/jit/forwardsub.cpp:486]: Bail up in `fgForwardSub`, and assert `fgDidEarlyLiveness' is true here?

**Review by AndyAyersMS — APPROVED** [review]
I'm fine with removing reliance on early ref counts.

File: src/coreclr/jit/forwardsub.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/forwardsub.cpp]: Makes sense, done

**Pull request merged**